google/cloud/spanner/client_test.cc

```diff
@@ -985,25 +973,18 @@ TEST(ClientTest, ProfileQuerySuccess) {
   auto rows = client.ProfileQuery(SqlStatement("SELECT * FROM Table;"));
 
   using RowType = std::tuple<std::string, std::int64_t>;
-  auto expected = std::vector<RowType>{
-      RowType("Ann", 42),
-  };
-  int row_number = 0;
-  for (auto& row : StreamOf<RowType>(rows)) {
-    ASSERT_LT(row_number, expected.size());
-    EXPECT_THAT(row, IsOkAndHolds(expected[row_number]));
-    ++row_number;
-  }
-  EXPECT_EQ(row_number, expected.size());
+  auto stream = StreamOf<RowType>(rows);
+  auto actual = std::vector<StatusOr<RowType>>{stream.begin(), stream.end()};
+  EXPECT_THAT(actual, ElementsAre(IsOkAndHolds(RowType("Ann", 42))));
+
   auto actual_plan = rows.ExecutionPlan();
   ASSERT_TRUE(actual_plan);
   EXPECT_THAT(*actual_plan, IsProtoEqual(stats.query_plan()));
 
   auto actual_stats = rows.ExecutionStats();
   ASSERT_TRUE(actual_stats);
-  std::unordered_map<std::string, std::string> expected_stats{
-      {"elapsed_time", "42 secs"}};
-  EXPECT_EQ(expected_stats, *actual_stats);
+  EXPECT_THAT(*actual_stats,
+              UnorderedElementsAre(Pair("elapsed_time", "42 secs")));
```

Comment:
I assume you have a reason to prefer `UnorderedElementsAre()` over `ElementsAre()` even though there is a single element?  If you have a minute can you share why?

Reply:
Yes ... what Darren said.